src/components/List.js

```diff
@@ -72,62 +111,131 @@ class List extends React.Component {
         }
     }
 
-    render() {
-        var callSize;
-        if (this.props.callSize === "large") {
-            callSize = {height: 60, width: 260};
-        } else {
-            callSize = {height: 50, width: 220};
+    static getPlaceholderIndex(props, NUMCOLUMNS, COLUMNSIZE) {
+        if (props.sort === "arrayOrder" && props.placeholderPosition) {
+            const {x, y, page} = props.placeholderPosition;
+            if (y < 0 || y >= COLUMNSIZE) {
+                return undefined;
+            } else if (x < 0 || x >= NUMCOLUMNS) {
+                return undefined;
+            } else {
+                const placeholderIndex = (y + COLUMNSIZE * (x + NUMCOLUMNS * (page)));
+                if (placeholderIndex < 0) {
+                    return undefined;
+                } else if (placeholderIndex > props.calls.length) {
+                    return props.calls.length;
+                } else {
+                    if (props.dragSourceOrigin && props.dragSourcePosition < placeholderIndex) {
+                        return placeholderIndex + 1;
+                    } else {
+                        return placeholderIndex;
+                    }
+                }
+            }
         }
+        return undefined;
+    }
 
-        const navHeight = 160;
-        var availableWidth, flexWidth;
-        if (this.props.size === "half") {
-            availableWidth = Math.min(this.props.windowWidth/2, 1200);
-            flexWidth = "col-md-6";
-        } else {
-            availableWidth = Math.min(this.props.windowWidth, 1300);
-            flexWidth = "col-md-12";
+    static handlePageTurns(props, timeout, NUMCOLUMNS, COLUMNSIZE) {
+        if (props.sort === "arrayOrder" && props.placeholderPosition) {
+            const {x, y} = props.placeholderPosition;
+            if (y < 0 || y >= COLUMNSIZE) {
+                clearTimeout(timeout);
+                return undefined;
+            } else if (x < 0) {
+                clearTimeout(timeout);
+                return setTimeout(() => {
+                    window.$(`#${props.id}`).carousel('prev');
+                }, 1000);
+            } else if (x >= NUMCOLUMNS) {
+                clearTimeout(timeout);
+                return setTimeout(() => {
+                    window.$(`#${props.id}`).carousel('next');
+                }, 1000);
+            } else {
+                clearTimeout(timeout);
+                return undefined;
+            }
+        } else if (timeout) {
+            clearTimeout(timeout);
         }
+        return undefined;
 
-        const NUMCOLUMNS = Math.floor((availableWidth-140)/callSize.width) || 1;
-        const COLUMNSIZE = Math.floor((this.props.windowHeight-navHeight)/callSize.height) || 1;
-        const sort = this.getSort();
+    }
+
+    render() {
+        const {NUMCOLUMNS, COLUMNSIZE, callSize, placeholderIndex, flexWidth} = this.state;
```

Comment:
I would also move a lot of this to a method so you can just return the objects after they are built.